integration/docker/csi/alluxio/nodeserver.go

```diff
@@ -182,37 +180,15 @@ func (ns *nodeServer) NodeStageVolume(ctx context.Context, req *csi.NodeStageVol
 	if err != nil {
 		return nil, err
 	}
-	if _, err := ns.client.CoreV1().Pods(os.Getenv("NAMESPACE")).Create(fusePod); err != nil {
-		if strings.Contains(err.Error(), "already exists") {
-			glog.V(4).Infof("Fuse pod %s already exists.", fusePod.Name)
-			return &csi.NodeStageVolumeResponse{}, nil
-		}
-		return nil, status.Errorf(codes.Internal, "Failed to launch Fuse Pod at %v.\n%v", ns.nodeId, err.Error())
-	}
-	glog.V(4).Infoln("Successfully creating Fuse pod.")
 
-	// Wait for alluxio-fuse pod finishing mount to global mount point
-	retry, err := strconv.Atoi(os.Getenv("FAILURE_THRESHOLD"))
-	if err != nil {
-		return nil, status.Errorf(codes.InvalidArgument, "Cannot convert failure threshold %v to int.", os.Getenv("FAILURE_THRESHOLD"))
-	}
-	timeout, err := strconv.Atoi(os.Getenv("PERIOD_SECONDS"))
-	if err != nil {
-		return nil, status.Errorf(codes.InvalidArgument, "Cannot convert period seconds %v to int.", os.Getenv("PERIOD_SECONDS"))
+	if err := ns.createFusePodIfNotExist(fusePod); err != nil {
+		return nil, err
```

Comment:
can we use the way of propagating error messages along the stream, so that we know the full stack trace of what's wrong in which step?

Reply:
CSI is the case that return values are much less important than logs. I made the logs more verbose than the return values.

Reply:
CSI doesn't use return error message?

Reply:
Is it possible to return error to the highest level (closest to client side), and then log the whole error message there, including all the stacktrace (all error message from the upper level)